apps/right-to-rent-check/behaviours/tenants.js

```diff
@@ -0,0 +1,30 @@
+'use strict';
+
+const _ = require('lodash');
+
+module.exports = fields => {
+
+  return superclass => class extends superclass {
+
+    saveValues(req, res, callback) {
+      const name = req.body['tenant-name'];
+      const tenants = req.sessionModel.get('tenants') || [];
+
+      let tenant = _.find(tenants, {'tenant-name': name});
+      if (!tenant) {
+        tenant = fields.length ? _.pick(req.body, fields) : req.body;
+        if (!_.isEmpty(tenant)) {
```

Comment:
Are we sure that the name is sufficiently unique? Is it possible to have multiple tenants with the same name?